iaac/terraform/aws-infra/ebs-csi-driver-irsa/main.tf

```diff
@@ -2,7 +2,7 @@ module "ebs_csi_driver_irsa" {
   source  = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
   version = "~> 5.20"
 
-  role_name_prefix = "${var.cluster_name}-${var.cluster_region}-ebs-csi-driver-"
+  role_name_prefix = substr("${var.cluster_name}-${var.cluster_region}-ebs-csi-driver-", 0, 38) # Can only be 38 characters or less
```

Comment:
is it better to override role_name https://github.com/terraform-aws-modules/terraform-aws-iam/blob/v5.29.0/modules/iam-role-for-service-accounts-eks/variables.tf#L7 since we are creating unique identifier?

looks like role_name_prefix helps create unique identifiers - https://registry.terraform.io/providers/figma/aws-4-49-0/latest/docs/resources/iam_role#name_prefix already. this is used in https://github.com/terraform-aws-modules/terraform-aws-iam/blob/v5.29.0/modules/iam-role-for-service-accounts-eks/main.tf#L71C3-L71C14
or we can shorten the prefix since tf will create unique identifier

Reply:
We can but cluster name + iam role name can be over 64 characters ie the iam role name limit, the same thing will have to be done.  I think the prefix approach is better: An eks cluster name has a [maximum length](https://docs.aws.amazon.com/eks/latest/APIReference/API_CreateCluster.html#AmazonEKS-CreateCluster-request-name) of 100, which is well over the iam role name limit of 64. If two clusters have the same characters uptil the 65th character the role name would be the same and a collision would occur.

Reply:
updated to substr("${var.cluster_region}-${var.cluster_name}-ebs-csi-driver", 0, 64) # Can only be 64 characters or less

Reply:
Sg, FYI, We restrict cluster name to 19 chars in our tf stacks for similar issues